### PR TITLE
Rewrite dotty to indent after #17522

### PR DESCRIPTION
**Pull request by adpi2**
Rewrite dotty to indent, using `-indent -rewrite`, to validate #17522.

### CODE REVIEW

File: compiler/src/dotty/tools/backend/jvm/BCodeBodyBuilder.scala [hunk 1]
```diff
@@ -111,192 +108,176 @@ trait BCodeBodyBuilder extends BCodeSkelBuilder {
...
 
     /* Generate code for primitive arithmetic operations. */
-    def genArithmeticOp(tree: Tree, code: Int): BType = tree match{
+    def genArithmeticOp(tree: Tree, code: Int): BType = tree match
       case Apply(fun @ DesugaredSelect(larg, _), args) =>
-      var resKind = tpeTK(larg)
+        var resKind = tpeTK(larg)
 
-      assert(resKind.isNumericType || (resKind == BOOL),
+        assert(resKind.isNumericType || (resKind == BOOL),
```
Comment (adpi2) [inline compiler/src/dotty/tools/backend/jvm/BCodeBodyBuilder.scala:120]: Some fixed indentation here and below in this file.

File: compiler/src/dotty/tools/backend/jvm/BCodeBodyBuilder.scala [hunk 2]
```diff
@@ -320,9 +301,8 @@ trait BCodeBodyBuilder extends BCodeSkelBuilder {
           else { genLoad(tree.rhs, tk) }
           bc.store(idx, tk)
           val localVarStart = currProgramPoint()
-          if (!isSynth) { // there are case <synthetic> ValDef's emitted by patmat
+          if (!isSynth) // there are case <synthetic> ValDef's emitted by patmat
```
Comment (adpi2) [inline compiler/src/dotty/tools/backend/jvm/BCodeBodyBuilder.scala:304]: a preserved comment

File: compiler/src/dotty/tools/backend/jvm/CoreBTypes.scala [hunk 3]
```diff
@@ -9,50 +9,49 @@ import scala.tools.asm.{Handle, Opcodes}
 import dotty.tools.dotc.core.StdNames
 import BTypes.InternalName
 
-abstract class CoreBTypes {
+abstract class CoreBTypes:
   val bTypes: BTypes
   import bTypes._
 
-   def primitiveTypeMap: Map[Symbol, PrimitiveBType]
+  def primitiveTypeMap: Map[Symbol, PrimitiveBType]
```
Comment (adpi2) [inline compiler/src/dotty/tools/backend/jvm/CoreBTypes.scala:16]: A fixed off-by-one indentation

File: compiler/src/dotty/tools/backend/sjs/JSCodeGen.scala [hunk 4]
```diff
@@ -1491,74 +1366,65 @@ class JSCodeGen()(using genCtx: Context) {
...
          * <undefined-param> that cannot be eliminated.
          * Such methods are never called anyway, because they are filtered out in
          * JSExportsGen.defaultGetterDenot().
          */
         None
-      } else /*if (sym.isClassConstructor && isHijackedBoxedClass(sym.owner)) {
+      else /*if (sym.isClassConstructor && isHijackedBoxedClass(sym.owner)) {
         None
-      } else*/ {
+      } else*/
```
Comment (adpi2) [inline compiler/src/dotty/tools/backend/sjs/JSCodeGen.scala:1391]: More preservation of comments

File: library/src/scala/Tuple.scala [hunk 5]
```diff
@@ -298,11 +272,9 @@ sealed trait NonEmptyTuple extends Tuple {
   inline def tail[This >: this.type <: NonEmptyTuple]: Tail[This] =
     runtime.Tuples.tail(this).asInstanceOf[Tail[This]]
 
-}
 
 @showAsInfix
 sealed abstract class *:[+H, +T <: Tuple] extends NonEmptyTuple
 
-object *: {
+object `*:`:
```
Comment (adpi2) [inline library/src/scala/Tuple.scala:279]: A back-quoted operator

**Comment by nicolasstucki** [comment]
@adpi2 do we still need this PR open? Or should we close it once https://github.com/lampepfl/dotty/pull/17522 is merged?

**Pull request closed without merging**